### DISCUSSION TITLE
Excellent repo

### DISCUSSION THREAD

**Discussion by peterhinch** [easytarget/microPyEZfonts#24] [General]
I've only just been made aware of this. Thank you for maintaining compatibility with [font-to-py](https://github.com/peterhinch/micropython-font-to-py) and consequently with my GUI's and display drivers.

Your RAM measurements are interesting. When I first conceived of `font-to-py` and `micro-gui` I was intent on running on an ESP8266, so minimising RAM use was a prime objective. With modern hardware your approach of using a `dict` makes sense and would have made my life a lot easier, especially when it came to implementing sparse fonts. As you have demonstrated, the API enables a compatible font format to be created with an entirely different `get_ch()` implementation.

**Comment by easytarget** [comment]
Thankyou Peter; it was really nice to read that; and apologies for not responding quickly. I was in Barlyman Butturburr mode and things kept driving it from my mind.

I was pleasantly surprised that the format worked as well as it did; as I mentioned elsewhere it originally came about because I'd written my own BDF parser to deal with some less-than-awesome fonts I wanted to import. The original plan was to grock and copy the binary format you use, but once I had all the data correctly padded and aligned it was just faster (lazier?) to convert to bytearrays and use as-is.

I think it is good that `get_ch()` is being used in multiple libraries and display subsystems. It's a very efficient function that does just what I needs to and no more, it's a good example for the motto: 'if it aint broke; dont fix it'.
And it has interesting possibilities; I have a WIP that on a 'font' that dynamically draws and then serves 7-segment display chars; the full hex numerics plus a couple of symbols; at an arbitary size you simply set in the file itself after copying to the device.
- After that all the various 'writer' classes should be able to just use it. 
- eg, `get_ch()` creates yet another framebuffer, draws the required char with scaled poly(s), then serves the buffer as memoryview.
- I want to add 'slant' and 'weight' options to this too, maybe (configurable) caching for characters to speed things up on dense displays.
 
I hope I can share soon but I have to throw myself into some Javasccript work (LaserWeb, love the program, regret the language) which is why I'm a bit more active right now; clearing the decks of outstanding MicroPython bits before I go off-course.
